# BÀI TẬP NHÓM LT PTDL1

Sử dụng dataset: ds_salaries.csv

## Phần I: Tìm hiểu về dataset

Import thư viện

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import os

Đọc dữ liệu từ file csv

In [2]:
df = pd.read_csv('../data/ds_salaries.csv')

Hiển thị toàn bộ dữ liệu

In [3]:
df

work_year experience_level employment_type                 job_title  \
0          2023               SE              FT  Principal Data Scientist   
1          2023               MI              CT               ML Engineer   
2          2023               MI              CT               ML Engineer   
3          2023               SE              FT            Data Scientist   
4          2023               SE              FT            Data Scientist   
...         ...              ...             ...                       ...   
3750       2020               SE              FT            Data Scientist   
3751       2021               MI              FT  Principal Data Scientist   
3752       2020               EN              FT            Data Scientist   
3753       2020               EN              CT     Business Data Analyst   
3754       2021               SE              FT      Data Science Manager   

       salary salary_currency  salary_in_usd employee_residence  remote_ratio  \
0       80000             EUR          85847                 ES           100   
1       30000             USD          30000                 US           100   
2       25500             USD          25500                 US           100   
3      175000             USD         175000                 CA           100   
4      120000             USD         120000                 CA           100   
...       ...             ...            ...                ...           ...   
3750   412000             USD         412000                 US           100   
3751   151000             USD         151000                 US           100   
3752   105000             USD         105000                 US           100   
3753   100000             USD         100000                 US           100   
3754  7000000             INR          94665                 IN            50   

     company_location company_size  
0                  ES            L  
1                  US            S  
2                  US            S  
3                  CA            M  
4                  CA            M  
...               ...          ...  
3750               US            L  
3751               US            L  
3752               US            S  
3753               US            L  
3754               IN            L  

[3755 rows x 11 columns]

Hiển thị 5 dòng dữ liệu đầu tiên

In [4]:
df.head()

work_year experience_level employment_type                 job_title  \
0       2023               SE              FT  Principal Data Scientist   
1       2023               MI              CT               ML Engineer   
2       2023               MI              CT               ML Engineer   
3       2023               SE              FT            Data Scientist   
4       2023               SE              FT            Data Scientist   

   salary salary_currency  salary_in_usd employee_residence  remote_ratio  \
0   80000             EUR          85847                 ES           100   
1   30000             USD          30000                 US           100   
2   25500             USD          25500                 US           100   
3  175000             USD         175000                 CA           100   
4  120000             USD         120000                 CA           100   

  company_location company_size  
0               ES            L  
1               US            S  
2               US            S  
3               CA            M  
4               CA            M

Hiển thị 5 dòng dữ liệu cuối cùng

In [5]:
df.tail()

work_year experience_level employment_type                 job_title  \
3750       2020               SE              FT            Data Scientist   
3751       2021               MI              FT  Principal Data Scientist   
3752       2020               EN              FT            Data Scientist   
3753       2020               EN              CT     Business Data Analyst   
3754       2021               SE              FT      Data Science Manager   

       salary salary_currency  salary_in_usd employee_residence  remote_ratio  \
3750   412000             USD         412000                 US           100   
3751   151000             USD         151000                 US           100   
3752   105000             USD         105000                 US           100   
3753   100000             USD         100000                 US           100   
3754  7000000             INR          94665                 IN            50   

     company_location company_size  
3750               US            L  
3751               US            L  
3752               US            S  
3753               US            L  
3754               IN            L

Mô tả về dataset

* work_year: Năm trả lương
* experience_level: Mức độ kinh nghiệm trong công việc với các giá trị tương ứng như sau:
    * EN: Entry-level / Junior: Người mới vào nghề
    * MI Mid-level / Intermediate: Người có vài năm kinh nghiệm
    * SE Senior-level/ Expert: Người có kinh nghiệm chuyên môn sâu sắc
    * EX Executive-level / Director: Người có kinh nghiệm lãnh đạo
* employment_type: Loại hình làm việc với các giá trị tương ứng như sau:
    * FT: Full-time: Toàn thời gian
    * PT: Part-time: Bán thời gian
    * CT: Contract: Hợp đồng
    * FL: Freelance: Làm việc tự do
* job_title: Vai trò đã đảm nhiệm trong năm
* salary: Lương thu nhập của nhân viên
* salary_currency: Đơn vị tiền tệ của tiền lương được trả cho nhân viên
* salary_in_usd: Mức lương bằng USD
* employee_residence: Quốc gia cư trú của nhân viên trong năm làm việc
* remote_ratio: Tổng khối lượng công việc được thực hiện từ xa trong năm làm việc, các giá trị tương ứng như sau:
    * 0: Không có công việc từ xa (dưới 20%)
    * 50: Có một phần công việc được làm từ xa
    * 100: Toàn bộ công việc được làm từ xa (hơn 80%)
* company_location: Quốc gia của trụ sở chính hoặc chi nhánh nơi làm việc của nhân viên
* company_size: Số người làm việc cho công ty trong năm, với các giá trị tương ứng như sau:
    * S: ít hơn 50 nhân viên (quy mô nhỏ)
    * M: từ 50 đến 250 nhân viên (quy mô trung bình)
    * L: nhiều hơn 250 nhân viên (quy mô lớn)

Sắp xếp dữ liệu theo cột 'Salary' giảm dần

In [7]:
df.sort_values(by='salary', ascending=False)

work_year experience_level employment_type  \
3669       2021               MI              FT   
3646       2020               MI              FT   
3574       2021               MI              FT   
3475       2021               MI              FT   
2966       2022               SE              FT   
...         ...              ...             ...   
3681       2020               MI              FT   
2589       2022               SE              FT   
2933       2022               MI              CT   
573        2023               EN              FT   
1548       2022               EN              FT   

                           job_title    salary salary_currency  salary_in_usd  \
3669                  Data Scientist  30400000             CLP          40038   
3646                  Data Scientist  11000000             HUF          35735   
3574                 BI Data Analyst  11000000             HUF          36259   
3475                     ML Engineer   8500000             JPY          77364   
2966  Lead Machine Learning Engineer   7500000             INR          95386   
...                              ...       ...             ...            ...   
3681                    Data Analyst      8000             USD           8000   
2589          Product Data Scientist      8000             USD           8000   
2933              Analytics Engineer      7500             USD           7500   
573    Autonomous Vehicle Technician      7000             USD           7000   
1548                    AI Developer      6000             EUR           6304   

     employee_residence  remote_ratio company_location company_size  
3669                 CL           100               CL            L  
3646                 HU            50               HU            L  
3574                 HU            50               US            L  
3475                 JP            50               JP            S  
2966                 IN            50               IN            L  
...                 ...           ...              ...          ...  
3681                 PK            50               PK            L  
2589                 IN           100               SG            L  
2933                 BO            50               BO            M  
573                  GH             0               GH            S  
1548                 MK             0               MK            S  

[3755 rows x 11 columns]

Chuyển kiểu dữ liệu cho 1 cột nào đó

In [8]:
df['salary'] = df['salary'].astype('float')

In [9]:
df['salary'].dtypes

dtype('float64')

Xem chiều dài của dataframe

In [11]:
# Xem chiều dài của dataframe
len(df)

3755

-> Kết quả trả về: Dataframe có 3755 dòng dữ liệu

Xem thông tin của dataframe

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3755 entries, 0 to 3754
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   work_year           3755 non-null   int64  
 1   experience_level    3755 non-null   object 
 2   employment_type     3755 non-null   object 
 3   job_title           3755 non-null   object 
 4   salary              3755 non-null   float64
 5   salary_currency     3755 non-null   object 
 6   salary_in_usd       3755 non-null   int64  
 7   employee_residence  3755 non-null   object 
 8   remote_ratio        3755 non-null   int64  
 9   company_location    3755 non-null   object 
 10  company_size        3755 non-null   object 
dtypes: float64(1), int64(3), object(7)
memory usage: 322.8+ KB


Xem kích thước của dataframe

In [13]:
df.shape

(3755, 11)

-> Kết quả trả về: Dataframe có 3755 dòng dữ liệu và 11 cột dữ liệu

Sử dụng hàm describe() để thống kê dữ liệu

In [14]:
df.describe()

work_year        salary  salary_in_usd  remote_ratio
count  3755.000000  3.755000e+03    3755.000000   3755.000000
mean   2022.373635  1.906956e+05  137570.389880     46.271638
std       0.691448  6.716765e+05   63055.625278     48.589050
min    2020.000000  6.000000e+03    5132.000000      0.000000
25%    2022.000000  1.000000e+05   95000.000000      0.000000
50%    2022.000000  1.380000e+05  135000.000000      0.000000
75%    2023.000000  1.800000e+05  175000.000000    100.000000
max    2023.000000  3.040000e+07  450000.000000    100.000000

Xem thống kê trên từng cột

In [15]:
# Xem thống kê trên từng cột
df.describe(include='all')

work_year experience_level employment_type      job_title  \
count   3755.000000             3755            3755           3755   
unique          NaN                4               4             93   
top             NaN               SE              FT  Data Engineer   
freq            NaN             2516            3718           1040   
mean    2022.373635              NaN             NaN            NaN   
std        0.691448              NaN             NaN            NaN   
min     2020.000000              NaN             NaN            NaN   
25%     2022.000000              NaN             NaN            NaN   
50%     2022.000000              NaN             NaN            NaN   
75%     2023.000000              NaN             NaN            NaN   
max     2023.000000              NaN             NaN            NaN   

              salary salary_currency  salary_in_usd employee_residence  \
count   3.755000e+03            3755    3755.000000               3755   
unique           NaN              20            NaN                 78   
top              NaN             USD            NaN                 US   
freq             NaN            3224            NaN               3004   
mean    1.906956e+05             NaN  137570.389880                NaN   
std     6.716765e+05             NaN   63055.625278                NaN   
min     6.000000e+03             NaN    5132.000000                NaN   
25%     1.000000e+05             NaN   95000.000000                NaN   
50%     1.380000e+05             NaN  135000.000000                NaN   
75%     1.800000e+05             NaN  175000.000000                NaN   
max     3.040000e+07             NaN  450000.000000                NaN   

        remote_ratio company_location company_size  
count    3755.000000             3755         3755  
unique           NaN               72            3  
top              NaN               US            M  
freq             NaN             3040         3153  
mean       46.271638              NaN          NaN  
std        48.589050              NaN          NaN  
min         0.000000              NaN          NaN  
25%         0.000000              NaN          NaN  
50%         0.000000              NaN          NaN  
75%       100.000000              NaN          NaN  
max       100.000000              NaN          NaN

Hiển thị kiểu dữ liệu của từng cột

In [16]:
df.dtypes

work_year               int64
experience_level       object
employment_type        object
job_title              object
salary                float64
salary_currency        object
salary_in_usd           int64
employee_residence     object
remote_ratio            int64
company_location       object
company_size           object
dtype: object